File: 2023/Leetcode/problem26.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        k = 1
        instances = 1
        while i + instances < len(nums):
            while i + instances < len(nums) and nums[i] == nums[i + instances]:
                instances += 1

            if i + instances < len(nums):
                nums[i + 1] = nums[i + instances]
                i += 1
                k += 1

        return k

File: 2024/Leetcode/problem26.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 // n: nums.length
 // Time Complexity: O(n) Linear Time
 // Space Complexity: O(1) Constant Space
var removeDuplicates = function(nums) {
    let red = 0;
    let black = 1;

    while (black < nums.length) {
        while (black < nums.length && nums[red] === nums[black]) {
            black++;
        }

        if (black < nums.length) {
            nums[red + 1] = nums[black];
            red++;
            black++;
        }
    }

    return red + 1;
};